doc/source/glossary.rst

```diff
@@ -0,0 +1,179 @@
+provide some commonality on how to refer to particular ideas within the field.
+
+A
+-
+.. _Address:
+    An address is another name for an :ref:`offset<Offset>`, specifically an offset within memory.  Offsets can be
+    both relative or absolute, whereas addresses are almost always absolute.
+
+.. _Address Space:
+
+Address Space
+    This is the name in volatility 2 for what's referred to as a :ref:`Translation Layer<translation layer>`.  It
+    encompasses all values that can be addresses, usually in reference to addresses in memory.
+
+.. _Alignment:
+
+Alignment
+    This value is what all data :ref:`offsets<offset>` will typically be a multiple of within a :ref:`type<type>`.
+
+.. _Array:
+
+Array
+    This represents a list of items, which can be access by an index, which is zero-based (meaning the first
+    element has index 0).  Items in arrays are almost always the same size (it is not a generic list, as in python)
+    even if they are :ref:`pointers<pointer>` to different sized objects.
+
+D
+-
+.. _Data Layer:
+
+Data Layer
+    A group of bytes, where each byte can be addressed by a specific offset.  Data layers are usually contiguous
+    chunks of data.
+
+.. _Dereference:
+
+Dereference
+    The act of taking the value of a pointer, and using it as an offset to another object, as a reference.
+
+.. _Domain:
+
+Domain
+    This the grouping for input values for a mapping or mathematical function.
+
+M
+-
+.. _Map:
+
+Map, mapping
+    A mapping is a relationship between two values, where one value (the :ref:`Domain<domain>` maps to the :ref:`Range<range>` value).
+    Mappings can be seen as a mathematical function, and therefore volatility 3 attempts to use mathematical functional
+    notation where possible.
+
+.. _Member:
+
+Member
+    The name of subcomponents of a type, similar to attributes of objects in common programming parlance.  These
+    are usually recorded as :ref:`offset<offset>` and :ref:`type<type>` pairs within a :ref:`structure<struct>`.
+
+O
+-
+.. _Object:
+
+Object
+    This has a specific meaning within computer programming (as in Object Oriented Programming), but within the world
+    of Volatility it is used to refer to a type that has been associated with a chunk of data.  See all :ref:`Type<type>`.
+
+.. _Offset:
+
+Offset
+    A numeric value that identifies a distance within a group of bytes, to uniquely identify a single byte, or the
+    start of a run of bytes.  This is often relative (offset from another object/item) but can be absolute (offset from
+    the start of a region of data).
+
+P
+-
+.. _Packed:
+
+Packed
+    Structures are often :ref:`aligned<alignment>` meaning that the various members (subtypes) are always aligned at
+    particular values (usually multiples of 2, 4 or 8).  Thus if a particular value is an odd number of bytes, the
+    next chunk of data containing useful information would start at an even offset, and a single byte of
```

Comment:
Its not so much about the value itself but rather about the length of the value, no?

>  Structures are often :ref:`aligned<alignment>` meaning that the various members (subtypes) have an offset that is a multiple of some number (usually multiples of 2, 4 or 8). Thus, if a particular member has a length which is not a multiple of the chosen number, there will be padding between it and the next member. In packed structs, no padding is used and the offset of the next member depends on the length of the previous one.

Reply:
Much more nicely worded, thanks!  5:D